integration/spark-common/src/main/scala/org/apache/spark/rdd/CarbonMergeFilesRDD.scala

```diff
@@ -124,17 +128,22 @@ class CarbonMergeFilesRDD(
       val split = theSplit.asInstanceOf[CarbonMergeFilePartition]
       logInfo("Merging carbon index files of segment : " +
               CarbonTablePath.getSegmentPath(tablePath, split.segmentId))
-
-      if (isHivePartitionedTable) {
-        CarbonLoaderUtil
-          .mergeIndexFilesInPartitionedSegment(carbonTable, split.segmentId,
-            segmentFileNameToSegmentIdMap.get(split.segmentId))
-      } else {
-        new CarbonIndexFileMergeWriter(carbonTable)
-          .mergeCarbonIndexFilesOfSegment(split.segmentId,
-            tablePath,
-            readFileFooterFromCarbonDataFile,
-            segmentFileNameToSegmentIdMap.get(split.segmentId))
+      try {
+        if (isHivePartitionedTable) {
+          CarbonLoaderUtil
+            .mergeIndexFilesInPartitionedSegment(carbonTable, split.segmentId,
+              segmentFileNameToSegmentIdMap.get(split.segmentId))
+        } else {
+          new CarbonIndexFileMergeWriter(carbonTable)
+            .mergeCarbonIndexFilesOfSegment(split.segmentId,
+              tablePath,
+              readFileFooterFromCarbonDataFile,
+              segmentFileNameToSegmentIdMap.get(split.segmentId))
+        }
+      } catch {
+        case e: Exception =>
+          LOGGER.warn(s"Failed to merge index files for ${ carbonTable.getTableUniqueName }. " +
```

Comment:
why the log level is warn, if the index merge failed, the unmerged index files also affect normally?

Reply:
yes if the merge index is not written then the index files would be used for scanning. This PR is just to ensure that the failure is writting merge index file should not affect the scanning or any subsequent load.